### PR TITLE
Autograph with make plxpr

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**

We have autograph functionality that allows us to convert user code with Python control flow to something PL-compatible, but it's not integrated in `make_plxpr`.

Also something happened with the changelog organization and things ended up under Qubit Mixed Device API and Dynamic Lie algebra that didn't belong there, so I shuffled things around.

**Description of the Change:**

We add this to `make_plxpr`. By default, autograph will be run on all functions. 

A few tests that were using `mocker.spy` had to be updated, because now `make_jaxpr` is called with an 'unknown' argument (the function that comes out of `run_autograph`), so we can't spy on its calls the same way we were.

[sc-71823]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -167,6 +170,8 @@ added `binary_mapping()` function to map `BoseWord` and `BoseSentence` to qubit
 * `qml.capture.qnode_call` has been made private and moved to the `workflow` module.
   [(#6620)](https://github.com/PennyLaneAI/pennylane/pull/6620/)
 
+* 
+
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:174]:
```suggestion
```

File: tests/capture/test_make_plxpr.py [hunk 2]
```diff
@@ -106,6 +112,101 @@ def circ():
...
+        assert "while_loop[" in str(plxpr2)
+
+        def eval(x):
+            return jax.core.eval_jaxpr(plxpr2.jaxpr, plxpr2.consts, x)
+
+        assert np.allclose(eval(0), [-1])
+        assert np.allclose(eval(5), [0])
+
+    def test_for_loop(self):
+        """Test that a foor loop is converted to a jaxpr with a ``while_loop`` function, and
```
Comment (JerryChen97) [inline tests/capture/test_make_plxpr.py:189]:
```suggestion
        """Test that a for loop is converted to a jaxpr with a ``for_loop`` function, and
```

File: tests/capture/test_make_plxpr.py [hunk 3]
```diff
@@ -106,6 +112,101 @@ def circ():
...
+
+            return qml.expval(qml.Z(0)), qml.expval(qml.Z(1))
+
+        dev = qml.device("default.qubit", wires=3)
+        qnode = qml.QNode(func, dev)
+
+        plxpr1 = qml.capture.make_plxpr(func)(jnp.array([0.0, 0.0]))
+        plxpr2 = qml.capture.make_plxpr(qnode)(jnp.array([0.0, 0.0]))
+
+        # the plxpr includes a representation of a `cond` function
```
Comment (JerryChen97) [inline tests/capture/test_make_plxpr.py:204]:
```suggestion
        # the plxpr includes a representation of a `for_loop` function
```

File: tests/capture/test_make_plxpr.py [hunk 4]
```diff
@@ -106,6 +112,101 @@ def circ():
...
+
+            return qml.expval(qml.Z(0))
+
+        dev = qml.device("default.qubit", wires=3)
+        qnode = qml.QNode(func, dev)
+
+        plxpr1 = qml.capture.make_plxpr(func)(0)
+        plxpr2 = qml.capture.make_plxpr(qnode)(0)
+
+        # the plxpr includes a representation of a `cond` function
```
Comment (JerryChen97) [inline tests/capture/test_make_plxpr.py:178]:
```suggestion
        # the plxpr includes a representation of a `while_loop` function
```

**Review by JerryChen97 — APPROVED** [review]
Thanks! Looks straightforward and good to me. Left some docstr suggestions

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -71,34 +71,12 @@
 added `binary_mapping()` function to map `BoseWord` and `BoseSentence` to qubit operators, using standard-binary mapping.
   [(#6564)](https://github.com/PennyLaneAI/pennylane/pull/6564)
 
-
-<h4>New API for Qubit Mixed</h4>
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:75]: This ended up in-between most of "new features" and the rest of "new features" - here its just moved further down

**Review by albi3ro — APPROVED** [review]
